### ISSUE TITLE
some concerns about the compilation on rpi4

### ISSUE THREAD

**Issue by Askmewho** [hrydgard/ppsspp#12318]
Hi! the problem are the next ones:
Ive targeted mesa libraries to compile ppsspp (libgl.so) and unutilizing glvndr libraries. the libgl lib has 2.1 gl and 3.0 ES gl, but for some reason it fails to execute. Ive tried preloading with library path a just gles sdl2 but its asking for glx support, and when I use my normal sdl (compiled by me with desktop gl include, it fail to create context...  which I dont get why, doesnt matter if ive set egl on or off. Ive tried every possibility, and for now I dont have another option than use my old pi3 build than targets libgl.so (2.1 gl). I will give you all more info about it.
maybe there are still conflicts between glvendor and mesa beside setting the preference, I will post the outputs here tomorrow!

thanks, Best emu of all time!!
masterpiece!

**Comment by unknownbrackets** [comment]
There were positive notes on the change in #12159.  Are you using the raspberry.armv7 toolchain and our cmake scripts, or are you using some custom / other build script?

Do other SDL apps work fine?

Have you tried USING_FBDEV and not?  I think we found that rpi4 shouldn't use FBDEV (which requires exiting the window manager to play the game), and instead can run windowed now.

-[Unknown]

**Comment by Askmewho** [comment]
Indeed, ive tried with and without fbdev! . I will try it again. Other sdl apps runs just fine, my old ppsspp compilation on rpi3 runs just fine on my pi4.. But its not a clean way, and ive cant stay on that version because i cant compile a new one with proper flags, etc.

**Comment by Askmewho** [comment]
so, for example, lets try egl, because ive had 3.0 ES

```
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
I: VulkanLoader.cpp:263: VulkanMayBeAvailable: Device allowed ('SDL:Linux')
I: VulkanLoader.cpp:270: VulkanMayBeAvailable: Library loaded ('libvulkan.so')
I: VulkanLoader.cpp:321: VulkanMayBeAvailable: Enumerating instance extensions
I: VulkanLoader.cpp:332: VulkanMayBeAvailable: Instance extension count: 2
I: VulkanLoader.cpp:340: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (00000009)
I: VulkanLoader.cpp:340: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000001)
E: VulkanLoader.cpp:358: Surface extension not found
DEBUG: Vulkan is not available, not using Vulkan.
[New Thread 0xa48c3000 (LWP 7585)]
I: Config.cpp:517: Longest display side: -1 pixels. Choosing scale 1
35:14:190 Core/Config.cpp:1187 I[LOADER]: Loading controller config: /home/pi/.config/ppsspp/PSP/SYSTEM/controls.ini
35:14:195 UI/NativeApp.cpp:387 E[LOADER]: Failed graphics backends: OPENGL,ALL,ALL,ALL
35:14:195 Core/Config.cpp:599 E[LOADER]: Graphics backend failed for 0, trying another
35:14:195 Core/Config.cpp:624 E[LOADER]: All graphics backends failed
35:14:204 Core/Config.cpp:1272 I[LOADER]: Config saved (got storage permission): '/home/pi/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
35:14:205 Core/Config.cpp:1285 I[LOADER]: Controller config saved: /home/pi/.config/ppsspp/PSP/SYSTEM/controls.ini
Pixels: 960 x 544
Virtual pixels: 960 x 544
OpenGL 2.0 or higher.
I: gpu_features.cpp:174: GPU Vendor : Broadcom ; renderer: V3D 4.2 version str: OpenGL ES 3.0 Mesa 19.1.0-devel ; GLSL version str: OpenGL ES GLSL ES 3.00
I: gpu_features.cpp:293: OpenGL ES 3.0 support detected!


Thread 1 "PPSSPPSDL" received signal SIGSEGV, Segmentation fault.
0x00000000 in ?? ()
(gdb) bt
#0  0x00000000 in ?? ()
#1  0x00430a64 in CheckGLExtensions() ()
#2  0x00413ae4 in SDLGLGraphicsContext::Init(SDL_Window*&, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) ()
#3  0x00049bb8 in main ()
(gdb)  c
Continuing.

[2]+  Detenido                gdb ./PPSSPPSDL
```

and with gles sdl...

```
pi@raspberrypi:~/Desktop/ppsspp/build $ LD_LIBRARY_PATH=/home/pi/Desktop/N64/gles ./PPSSPPSDL 
I: VulkanLoader.cpp:263: VulkanMayBeAvailable: Device allowed ('SDL:Linux')
I: VulkanLoader.cpp:270: VulkanMayBeAvailable: Library loaded ('libvulkan.so')
I: VulkanLoader.cpp:321: VulkanMayBeAvailable: Enumerating instance extensions
I: VulkanLoader.cpp:332: VulkanMayBeAvailable: Instance extension count: 2
I: VulkanLoader.cpp:340: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (00000009)
I: VulkanLoader.cpp:340: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000001)
E: VulkanLoader.cpp:358: Surface extension not found
DEBUG: Vulkan is not available, not using Vulkan.
I: Config.cpp:517: Longest display side: -1 pixels. Choosing scale 1
36:28:529 Core/Config.cpp:1187 I[LOADER]: Loading controller config: /home/pi/.config/ppsspp/PSP/SYSTEM/controls.ini
36:28:533 UI/NativeApp.cpp:387 E[LOADER]: Failed graphics backends: OPENGL,ALL,ALL,ALL,ALL
36:28:533 Core/Config.cpp:599 E[LOADER]: Graphics backend failed for 0, trying another
36:28:533 Core/Config.cpp:624 E[LOADER]: All graphics backends failed
36:28:542 Core/Config.cpp:1272 I[LOADER]: Config saved (got storage permission): '/home/pi/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
36:28:543 Core/Config.cpp:1285 I[LOADER]: Controller config saved: /home/pi/.config/ppsspp/PSP/SYSTEM/controls.ini
Pixels: 960 x 544
Virtual pixels: 960 x 544
SDL_CreateWindow failed for GL 4.6: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.5: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.4: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.3: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.2: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.1: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 4.0: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 3.3: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 3.2: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 3.1: SDL not configured with OpenGL/GLX support
SDL_CreateWindow failed for GL 3.0: SDL not configured with OpenGL/GLX support
36:28:552 Core/Config.cpp:1272 I[LOADER]: Config saved (NativeShutdown): '/home/pi/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
36:28:553 Core/Config.cpp:1285 I[LOADER]: Controller config saved: /home/pi/.config/ppsspp/PSP/SYSTEM/controls.ini
I: NativeApp.cpp:1308: NativeShutdown called
SDL_CreateWindow failed: SDL not configured with OpenGL/GLX support
GL init error ''
E: SDLMain.cpp:587: Output audio samples: 1024 (requested: 2048)
loading control pad mappings from gamecontrollerdb.txt: SUCCESS!
I: NativeApp.cpp:776: NativeInitGraphics
```

**Comment by unknownbrackets** [comment]
It seems like the second version was not compiled with USING_GLES2.  Note that this flag is actually for "GLES2+" as compared to desktop GL.  It's not to prevent use of GLES3 or anything.

The first version, it seems like we ended up with a GLES driver and tried to run with it.  It seems like glGetIntegerv crashed, though, possibly because we loaded using glew and it didn't understand ES?

I'd definitely suggest trying with USING_GLES2, which I'd expect to make GLES3 work given these two logs.

-[Unknown]
